modules/claims_api/app/workers/claims_api/special_issue_updater.rb

```diff
@@ -0,0 +1,146 @@
+    # @param special_issues [Array(String)] List of special issues to append
+    def perform(user, contention_id, special_issues)
+      validate_contention_id_structure(contention_id)
+      service = bgs_service(user).contention
+
+      claims = service.find_contentions_by_ptcpnt_id(user.participant_id)[:benefit_claims]
+      claim = claim_from_contention_id(claims, contention_id)
+      raise "Claim not found with contention: #{contention_id}" if claim.blank?
+
+      options = required_claim_fields(claim, contention_id, special_issues)
+      service.manage_contentions(options)
+    rescue BGS::ShareError, BGS::PublicError => e
+      # TODO: https://vajira.max.gov/browse/API-3277
+      log_exception_to_sentry(e)
+    end
+
+    # Service object to interface with BGS
+    #
+    # @param user [OpenStruct] Veteran to attach special issues to
+    # @return [BGS::Services] Service object
+    def bgs_service(user)
+      external_key = user.common_name || user.email
+
+      BGS::Services.new(
+        external_uid: user.icn || user.uuid,
+        external_key: external_key
+      )
+    end
+
+    # Find matching claim within all claims for the particular participant
+    #
+    # @param claims [Array(Hash)] List of BGS claim objects
+    # @param contention_id [Hash(claim_id:, code:, name:)] Identifier to match existing contention
+    # @return [Hash] Particular BGS claim object if found, nil if not found
+    def claim_from_contention_id(claims, contention_id)
+      claims.each do |claim|
+        next if claim[:contentions].blank?
+
+        contention = claim[:contentions].find { |c| matches_contention?(contention_id, c) }
+        return claim if contention.present?
+      end
+    end
+
+    # Generate expected payload for updating special issues through BGS
+    #
+    # @param claim [Hash] BGS claim object
+    # @param contention_id [Hash(claim_id:, code:, name:)] Identifier to match existing contention
+    # @param special_issues [Array(String)] List of special issues to append
+    # @return [Hash] Expected payload to send to BGS
+    def required_claim_fields(claim, contention_id, special_issues)
+      {
+        jrn_dt: claim[:jrn_dt],
+        bnft_clm_tc: claim[:bnft_clm_tc],
+        bnft_clm_tn: claim[:bnft_clm_tn],
+        claim_rcvd_dt: claim[:claim_rcvd_dt],
+        clm_id: claim[:clm_id],
+        lc_stt_rsn_tc: claim[:lc_stt_rsn_tc],
+        lc_stt_rsn_tn: claim[:lc_stt_rsn_tn],
+        lctn_id: claim[:lctn_id],
+        non_med_clm_desc: claim[:non_med_clm_desc],
+        prirty: claim[:prirty],
+        ptcpnt_id_clmnt: claim[:ptcpnt_id_clmnt],
+        ptcpnt_id_vet: claim[:ptcpnt_id_vet],
+        ptcpnt_suspns_id: claim[:ptcpnt_suspns_id],
+        soj_lctn_id: claim[:soj_lctn_id],
+        contentions: existing_contentions(claim, contention_id, special_issues)
+      }
+    end
+
+    # Generate expected contentions payload.
+    #
+    # @param claim [Hash] BGS claim object
```

Comment:
I'm okay with this style of annotation as long as we are consistent in implementing it in all code moving forward, probably something for us to bring up in our team agreement meeting next week.